src/plugins/expressions/common/execution/execution.ts

```diff
@@ -196,17 +203,47 @@ export class Execution<
         return createError({ message: `Function ${fnName} could not be found.` });
       }
 
+      let args: Record<string, ExpressionValue> = {};
+      let timeStart: number | undefined;
+
       try {
-        // Resolve arguments before passing to function
-        // resolveArgs returns an object because the arguments themselves might
-        // actually have a 'then' function which would be treated as a promise
+        // `resolveArgs` returns an object because the arguments themselves might
+        // actually have a `then` function which would be treated as a `Promise`.
         const { resolvedArgs } = await this.resolveArgs(fnDef, input, fnArgs);
+        args = resolvedArgs;
+        timeStart = this.params.debug ? Date.now() : 0;
```

Comment:
Wonder if makes to consider using `performance.now` + it's node analogue for this

Reply:
Done. Good idea.